### PR TITLE
more doctests to Reducible

### PR DESCRIPTION
**Pull request by dseer0**
added few doctests to reducible, 

its my first pr, im open to comments and suggestions

### CODE REVIEW

File: core/src/main/scala/cats/Reducible.scala [hunk 1]
```diff
@@ -44,13 +44,32 @@ import simulacrum.{noop, typeclass}
...
     reduce(fga)(G.algebra)
 
   /**
    * Apply `f` to each element of `fa` and combine them using the
    * given `Semigroup[B]`.
+   *
+   * {{{
+   * scala> import cats.Reducible
+   * scala> import cats.data.NonEmptyList
+   * scala> import cats.implicits._
```
Comment (armanbilge) [inline core/src/main/scala/cats/Reducible.scala:65]: The `cats.implicits._` import is now deprecated.

```suggestion
   * scala> import cats.syntax.all._
```

**Review by armanbilge — COMMENTED** [review]
Thanks, looks good! Just one very tiny thing :)

**Comment by dseer0** [comment]
checked it locally after commit (unlucky) and it failed with: Unused import cats.syntax.all._
apparently I dont even need it (not sure why it didnt fail with implicits._ : p)

**Comment by dseer0** [comment]
@armanbilge is it ok if I 'fix' other tests in this file? or it should be done with another pull request?

**Comment by armanbilge** [comment]
Sure, that would be great, thank you very much!

**Comment by dseer0** [comment]
@armanbilge if you could please rerun CI in freetime : )

**Comment by armanbilge** [comment]
Woops, sorry about that! Feel free to ping me if that happens again.

**Review by armanbilge — APPROVED** [review]
Looks good, thanks again!

**Review by rossabaker — APPROVED** [review]

**Pull request merged**